src/history/hs.c

```diff
@@ -212,54 +211,32 @@ __wt_hs_destroy(WT_SESSION_IMPL *session)
  *     Open a new history store table cursor.
  */
 int
-__wt_hs_cursor_open(WT_SESSION_IMPL *session)
+__wt_hs_cursor_open(WT_SESSION_IMPL *session, uint32_t *session_flags)
 {
     WT_CURSOR *cursor;
     WT_DECL_RET;
     const char *open_cursor_cfg[] = {WT_CONFIG_BASE(session, WT_SESSION_open_cursor), NULL};
 
+    /* Not allowed to open a cursor if you already have one */
+    WT_ASSERT(session, session->hs_cursor == NULL && !F_ISSET(session, WT_SESSION_HS_CURSOR));
+
     WT_WITHOUT_DHANDLE(
       session, ret = __wt_open_cursor(session, WT_HS_URI, NULL, open_cursor_cfg, &cursor));
     WT_RET(ret);
 
     /* History store cursors should always ignore tombstones. */
     F_SET(cursor, WT_CURSTD_IGNORE_TOMBSTONE);
 
-    session->hs_cursor = cursor;
-    F_SET(session, WT_SESSION_HS_CURSOR);
-
-    return (0);
-}
-
-/*
- * __wt_hs_cursor --
- *     Return a history store cursor, open one if not already open.
- */
-int
-__wt_hs_cursor(WT_SESSION_IMPL *session, uint32_t *session_flags, bool *is_owner)
-{
     /*
-     * We don't want to get tapped for eviction after we start using the history store cursor; save
-     * a copy of the current eviction state, we'll turn eviction off before we return.
-     *
-     * Don't cache history store table pages, we're here because of eviction problems and there's no
-     * reason to believe history store pages will be useful more than once.
+     * We don't want to get tapped for eviction after we start using the history store cursor. Save
+     * a copy of the current flag values. We'll restore them when the cursor is closed.
      */
     *session_flags = F_MASK(session, WT_HS_SESSION_FLAGS);
-    *is_owner = false;
-
-    /* Open a cursor if this session doesn't already have one. */
-    if (!F_ISSET(session, WT_SESSION_HS_CURSOR)) {
-        /* The caller is responsible for closing this cursor. */
-        *is_owner = true;
-        WT_RET(__wt_hs_cursor_open(session));
-    }
-
-    WT_ASSERT(session, session->hs_cursor != NULL);
-
-    /* Configure session to access the history store table. */
     F_SET(session, WT_HS_SESSION_FLAGS);
 
+    session->hs_cursor = cursor;
+    F_SET(session, WT_SESSION_HS_CURSOR);
```

Comment:
Were we able to figure out if WT_SESSION_HS_CURSOR flag is redundant for handling operations when the prepared state is set? I forgot what was the conclusion of the conversation from slack messages.

Reply:
Yeah, {{WT_SESSION_HS_CURSOR}} doesn't seem to be needed there.  I've addressed that as part of WT-6463.  I'll be merging WT-6463 after I get this one merged.  If you're curious about how that looks, feel free to review [the PR for WT-6463.](https://github.com/wiredtiger/wiredtiger/pull/5886) 😀